src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py

```diff
@@ -395,26 +397,27 @@ def calculate(
             # Therefore, the variables are named with a `_x100` suffix.
             usable_capacity_x100 = capacity * (soc_upper_bound - soc_lower_bound)
             soc_scaled = (
-                (soc - soc_lower_bound) / (soc_upper_bound - soc_lower_bound) * 100
+                (soc - soc_lower_bound) / (soc_upper_bound - soc_lower_bound) * 100.0
             )
             # we are clamping here because the SoC might be out of bounds
-            soc_scaled = min(max(soc_scaled, 0), 100)
+            soc_scaled = min(max(soc_scaled, 0.0), 100.0)
             timestamp = max(timestamp, metrics.timestamp)
             used_capacity_x100 += usable_capacity_x100 * soc_scaled
             total_capacity_x100 += usable_capacity_x100
 
         if timestamp == _MIN_TIMESTAMP:
             return Sample(datetime.now(tz=timezone.utc), None)
 
+        pct = 0.0
```

Comment:
I would add a comment about the intention of `pct` being literally `0.0` and `100.0` on the edges, just in case anyone feels tempted to *simplify* the code.